src/runtime_src/core/edge/ps_kernels/built_in/instance_query/instance_query.c

```diff
@@ -0,0 +1,272 @@
+          val.erase(val.size()-1);
+        }
+        os_pt.put("distribution", val);
+      }
+    }
+    ifs.close();
+
+    std::string model("unknown");
+    std::ifstream stream("/proc/device-tree/model");
+    if (stream.good()) {
+      std::getline(stream, model);
+      stream.close();
+    }
+    os_pt.put("model", model);
+
+    os_pt.put("cores", std::thread::hardware_concurrency());
+    os_pt.put("address_space", (boost::format("0x%lx") % (sysconf(_SC_PHYS_PAGES) * sysconf(_SC_PAGE_SIZE))).str());
+    get_mem_info(os_pt);
+    pt.add_child("os", os_pt);
+}
+
+static void
+get_instance_process_status( const std::string& pid,
+                    pt::ptree &pt)
+{
+  // Format the file path to point to a process status area
+  std::string file_path = "/proc/" + pid + "/status";
+  pt::ptree data_pt;
+  parse_file(data_pt, file_path, false);
+
+  // Remove undesired nodes
+  // Lambda to detect unwanted node names
+  auto inv_name = [](const std::pair<std::string, pt::ptree>& item) {
+    // The list of node names that should be removed
+    static std::set<std::string> invalid_names = {"Name"};
+    return invalid_names.find(item.second.get<std::string>("name")) != invalid_names.end(); 
+  };
+
+  // Shorthand for find_if function
+  auto find_invalid = [&](pt::ptree& p_pt) {
+    return std::find_if(p_pt.begin(), p_pt.end(), inv_name);
+  };
+
+  // Loop until all invalid names are removed
+  for (auto it = find_invalid(data_pt); it != data_pt.end(); it = find_invalid(data_pt))
+    data_pt.erase(it);
+
+  pt.add_child("process_info", data_pt);
+}
+
+static void
+get_instance_status(const std::string& file,
+                    pt::ptree &pt)
+{
+  std::string pid_path = file + "/status";
+  pt::ptree status_pt;
+  parse_file(status_pt, pid_path);
+
+  static std::map<std::string, std::string> name_map = {
+      { "PID", "pid"}
+  };
+  filter_ptree_contents(pt, status_pt, name_map);
+}
+
+static void
+get_instance_info(  const std::string& file,
+                    pt::ptree &pt)
+{
+  std::string info_path = file + "/cu_info";
+  pt::ptree info_pt;
+  parse_file(info_pt, info_path);
+
+  static std::map<std::string, std::string> name_map = {
+      { "Kernel name",           "kernel"               },
+      { "Instance(CU) name",     "name"                 },
+      { "CU address",            "cu_address"           },
+      { "CU index",              "cu_index"             },
+      { "Protocol",              "protocol"             },
+      { "Interrupt cap",         "interrupt_compatible" },
+      { "SW Resettable",         "resettable"           },
+      { "Number of arguments",   "argument_count"       }
+  };
+  filter_ptree_contents(pt, info_pt, name_map);
+}
+
+__attribute__((visibility("default")))
+int get_ps_kernel_data(char *output, int count, struct xrtHandles *xrtHandle)
+{
+  openlog("new_kernel_source", LOG_PID | LOG_CONS | LOG_NDELAY, LOG_NEWS);
+  syslog(LOG_INFO, "%s: Started new kernel\n", __func__);
```

Comment:
Maybe you can add a new argument for debug level, and only print to syslog when debug level is enabled.
This will then only print information for debugging purpose when necessary